jenkins/spark-premerge-build.sh

```diff
@@ -155,6 +155,12 @@ ci_2() {
     # Download a Scala 2.12 build of spark
     prepare_spark $SPARK_VER 2.12
     ./integration_tests/run_pyspark_from_build.sh
+
+    # Test a portion of cases for non-UTC time zone because of limited GPU resources.
+    # Here testing: parquet scan, orc scan, csv scan, cast, TimeZoneAwareExpression, FromUTCTimestamp
+    # Nightly CIs will cover all the cases.
+    TZ=Iran TEST='test_parquet_read_round_trip or test_read_round_trip or test_basic_csv_read or test_cast_string_ts_valid_format or test_unix_timestamp or test_from_utc_timestamp' ./integration_tests/run_pyspark_from_build.sh
```

Comment:
I just renamed the title with "databricks" postfix and re-trigger a pre-merge. Let's ensure we don't break databrick shim before merging this PR.

Reply:
it would not, databricks does not run this script. instead its relying 
on a dedicated test script https://github.com/NVIDIA/spark-rapids/blob/branch-23.12/jenkins/databricks/test.sh

Reply:
Yes, let's double confirm DBX runtime was passed. Hopefully I am over worried on this.